include/glow/Optimizer/GraphOptimizer/GraphOptimizer.h

```diff
@@ -29,12 +30,30 @@ class Module;
 class PlaceholderBindings;
 class Placeholder;
 
+/// \returns the default graph optimization pipeline.
+FunctionPassPipeline getDefaultGraphOptimizationPipeline();
+
+/// \returns the graph optimization pipeline for target independent
+/// optimizations.
+FunctionPassPipeline getTargetIndependentGraphOptimizationPipeline();
+
+/// \returns the default folding pipeline.
+FunctionPassPipeline getDefaultFoldPipeline();
+
 /// Perform optimizations on the graph representation.
-void optimize(Function *F, CompilationContext &cctx);
-void optimize(Function *F, CompilationMode mode);
-/// Fold nodes that were expressed lowered in the input model.
-void fold(Function *F, CompilationContext &cctx);
-void fold(Function *F, CompilationMode mode);
+void optimize(Function *F, CompilationContext &cctx,
```

Comment:
> CompilationContext &cctx,

this reads as optimize would change the cctx. can that be `const`?

Reply:
It was dropped by @ZchiPitt in #3032, as they thought they were going to add the log to the cctx and modify it. But then that went in a different direction. So we could add it back.

Reply:
Ok, I see backend might modify those params.